src/smartContracts/smartContract.ts

```diff
@@ -64,7 +64,7 @@ export class SmartContract {
       target: this.address,
       ...options,
     }
-    return this.provider.readSC(readParams)
+    return this.provider.readSC(readParams) as Promise<ReadSCData>
```

Comment:
this cast could be removed if addressing the previous comment

Reply:
so we can remove the cast ? ^^

Reply:
my bad yes